packages/core-data/src/hooks/use-resource-permissions.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Internal dependencies
+ */
+import { store as coreStore } from '../';
+import { Status } from './constants';
+import useQuerySelect from './use-query-select';
+
+interface GlobalResourcePermissionsResolution {
+	/** Can the current user create new resources of this type? */
+	canCreate: boolean;
+}
+interface SpecificResourcePermissionsResolution {
+	/** Can the current user update resources of this type? */
+	canUpdate: boolean;
+	/** Can the current user delete resources of this type? */
+	canDelete: boolean;
+}
+interface ResolutionDetails {
+	/** Resolution status */
+	status: Status;
+	/**
+	 * Is the data still being resolved?
+	 */
+	isResolving: boolean;
+}
+
+/**
+ * Is the data resolved by now?
+ */
+type HasResolved = boolean;
+
+type ResourcePermissionsResolution< IdType > = [
+	HasResolved,
+	ResolutionDetails &
+		GlobalResourcePermissionsResolution &
+		( IdType extends void ? SpecificResourcePermissionsResolution : {} )
+];
+
+/**
+ * Resolves resource permissions.
+ *
+ * @param  resource The resource in question, e.g. media.
+ * @param  id       ID of a specific resource entry, if needed, e.g. 10.
+ *
+ * @example
+ * ```js
+ * import { useResourcePermissions } from '@wordpress/core-data';
+ *
+ * function PagesList() {
+ *   const { canCreate, isResolving } = useResourcePermissions( 'pages' );
+ *
+ *   if ( isResolving ) {
+ *     return 'Loading ...';
+ *   }
+ *
+ *   return (
+ *     <div>
+ *       {canCreate ? (<button>+ Create a new page</button>) : false}
+ *       // ...
+ *     </div>
+ *   );
+ * }
+ *
+ * // Rendered in the application:
+ * // <PagesList />
+ * ```
+ *
+ * In the above example, when `PagesList` is rendered into an
+ * application, the appropriate permissions and the resolution details will be retrieved from
+ * the store state using `canUser()`, or resolved if missing.
+ *
+ * @return Entity records data.
+ * @template IdType
+ */
+export default function __experimentalUseResourcePermissions< IdType = void >(
+	resource: string,
+	id: IdType
```

Comment:
I guess we should make it optional in type?

```suggestion
	id?: IdType
```

(Not sure if it's needed)

Reply:
Good call!